### ISSUE TITLE
hakchi could workaround "jumpy" NES Classic menu behavior with less than 12 items in a folder be duplicating items when uploading config

### ISSUE THREAD

**Issue by The-MAZZTer** [ClusterM/hakchi2#416]
Issues #174 and #349 are both reports of this issue closed as won't fix. Despite it being a flaw in the original software, it's not a bug you see in supported usage of the NES Classic. So it really falls on us to either ignore it (the official stance so far, I consider it a legitimate position to take) fix it (if possible) or find a workaround, and I think I have one.

Hakchi2 could workaround this issue transparently by duplicating entries as needed until there are at least 12 items. Then the carousel works as it does for the 30 games. The only downside is if you have thumbnails on they get duplicated... but I think that's a very minor issue.

Items would not be duplicated in the hakchi2 UI at all and would only be duplicated when generating and saving the config for the NES Classic.

Possible pseudocode:

When writing the configuration for any folder of items (including back button, if any):
```
If the folder has more than 0 items:
    while the item list has less than 12 items:
        copy each of the ORIGINAL items in the list and append them to the current working item list
iterate through the working item list and generate configuration entries as normal
discard the modified item list once configuration of the folder is complete, as it's only used here.
```

**Comment by ClusterM** [comment]
Good idea. But we need to duplicate save-states too somehow.

**Comment by The-MAZZTer** [comment]
Oh, that could be a deal breaker if you have to do that. I assumed the entries were just pointers to games and you could have multiple ones that pointed to the same game. If that is not the case my idea might not work.

Possibly you could use hard links in some way to make a "copy" of a game that just referenced the same ROM file and save states and so forth.

**Comment by madmonkey1907** [comment]
"duplicate" folder could contain just the *.desktop file - no need to edit this file even. remember, it's blindly executed. save states will be in the same place.

**Comment by ClusterM** [comment]
@madmonkey1907 no, save-states file naming is based on game ID which can't be the same.

**Comment by madmonkey1907** [comment]
I don't remember actually :) what happens if id is the same?
you can use softlinks (ln -s), but there it starts to be more complicated than what it's worth.

**Comment by ClusterM** [comment]
AFAIK, id must be the same as directory name, so it can't be the same for different games.
Yes, it's possible use softlinks in theory.
